### PR TITLE
feat(platform): don't allow freezing non existent identity

### PR DESCRIPTION
**Pull request by QuantumExplorer**
## Issue being fixed or feature implemented
Added a new error type `IdentityToFreezeDoesNotExistError` to handle cases where an identity that is supposed to be frozen does not exist.

## What was done?
- Introduced `IdentityToFreezeDoesNotExistError` in the consensus error codes.
- Updated the `StateError` enum to include the new error.
- Implemented checks in the token freeze transition validation to utilize this new error when an identity to be frozen is not found.

## How Has This Been Tested?
- Added unit tests to ensure that the new error is raised correctly when attempting to freeze a non-existent identity.
- Existing tests for token freeze functionality were updated to account for this new validation.

## Breaking Changes
None

## Checklist
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

## Summary by CodeRabbit

- **New Features**
  - Added a new error type for cases when an identity to be frozen does not exist.
  - Introduced explicit validation for identity existence during token freeze and transfer operations, returning specific errors if identities are missing.
  - Enabled versioned validation methods for checking identity and non-masternode identity existence in state transitions.
  - Added helper functions for efficient identity existence validation and retrieval.

- **Bug Fixes**
  - Improved error handling and reporting for token freeze and transfer actions involving non-existent identities.

- **Tests**
  - Added extensive tests covering token burning, configuration updates, freezing (including non-existent identities), and transfers with various edge cases.

- **Refactor**
  - Streamlined identity existence checks in state transition validations using new helper functions to improve consistency and maintainability.

- **Chores**
  - Extended platform versioning and validation flags to support new identity existence checks.
  - Enhanced random identifier generation for testing purposes.

### CODE REVIEW

File: packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/data_contract_create/state/v0/mod.rs [hunk 1]
```diff
@@ -85,24 +83,15 @@ impl DataContractCreateStateTransitionStateValidationV0 for DataContractCreateTr
...
-                    execution_context.add_operation(ValidationOperation::RetrieveIdentity(
-                        RetrieveIdentityInfo::one_key(),
-                    ));
+                    let identity_exists = validate_non_masternode_identity_exists(
+                        platform.drive,
+                        member_identity_id,
+                        execution_context,
+                        tx,
+                        platform_version,
+                    )?;
```
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/data_contract_create/state/v0/mod.rs:93]: it's okay

File: packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/data_contract_update/state/v0/mod.rs [hunk 2]
```diff
@@ -174,24 +172,15 @@
...
-                    ));
-
-                    if maybe_key.is_none() {
+                    let identity_exists = validate_non_masternode_identity_exists(
+                        platform.drive,
+                        &member_identity_id,
+                        execution_context,
+                        tx,
+                        platform_version,
+                    )?;
```
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/data_contract_update/state/v0/mod.rs:181]: good catch

File: packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/data_contract_update/state/v0/mod.rs [hunk 3]
```diff
@@ -280,16 +260,15 @@ impl DataContractUpdateStateTransitionStateValidationV0 for DataContractUpdateTr
                         distribution.distribution_recipient()
                     {
                         if !validated_identities.contains(&identifier) {
-                            let maybe_balance = platform.drive.fetch_identity_balance(
-                                identifier.to_buffer(),
+                            let identity_exists = validate_identity_exists(
+                                platform.drive,
+                                &identifier,
+                                execution_context,
```
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/data_contract_update/state/v0/mod.rs:266]: no, it's okay

**Review by pauldelucia — APPROVED** [review]
